core-rust/core-api-server/src/core_api/handlers/state_non_fungible.rs

```diff
@@ -40,21 +41,22 @@ pub(crate) async fn handle_state_non_fungible(
         .into_response_error("non_fungible_id"));
     }
 
-    let non_fungible = read_optional_collection_substate::<Option<ScryptoRawValue>>(
+    // TODO(during review): I believe I changed the behavior here (previously we returned error for
+    // deleted substate, and now we will return `is_deleted=true`); is this the right thing to do?
```

Comment:
Will need someone from network team to help with this. :)

Reply:
This seems reasonable to me 👍 

So we can return "does not exist", "exists" or "is deleted". Sweet 👍